recipes/gobject-introspection/all/conanfile.py

```diff
@@ -0,0 +1,112 @@
+from conans import ConanFile, tools, Meson, VisualStudioBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+import os
+import shutil
+import glob
+
+
+class GobjectIntrospectionConan(ConanFile):
+    name = "gobject-introspection"
+    description = "GObject introspection is a middleware layer between C libraries (using GObject) and language bindings"
+    topics = ("conan", "gobject-instrospection")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://gitlab.gnome.org/GNOME/gobject-introspection"
+    license = "LGPL-2.1"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    generators = "pkg_config"
+
+    @property
+    def _is_msvc(self):
+        return self.settings.compiler == "Visual Studio"
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
```

Comment:
```suggestion
            del self.options.fPIC

    def validate(self):
```

Reply:
I had very strange bugs on CCI when using validate(self) the last days:
eg https://github.com/conan-io/conan-center-index/pull/4181#issuecomment-756993143 or https://github.com/conan-io/conan-center-index/pull/4182#issuecomment-757434081
so I'm not very inclined to use it until we understand what happened.  Can we wait to introduce it ?

Reply:
#4193 too

Reply:
Fair enough, `validate` is experimental and even missing from the Conan changelog :) Hope we would refactor recipes to use `validate` later.